src/test/java/com/google/cloud/spanner/pgadapter/JdbcSimpleModeMockServerTest.java

```diff
@@ -172,83 +177,327 @@ public void testQueryWithParameters() throws SQLException {
-            String.format(
-                "ERROR: INVALID_ARGUMENT: Statement type is not supported for batching \"%s\"",
-                sql.substring(0, sql.length() - 1)),
-            exception.getMessage());
+        assertThat(
+            exception.getMessage(), containsString("INVALID_ARGUMENT: Statement is invalid."));
       }
     }
+
+    List<ExecuteSqlRequest> requests = mockSpanner.getRequestsOfType(ExecuteSqlRequest.class);
+    assertEquals(4, requests.size());
+    assertEquals(INSERT_STATEMENT.getSql(), requests.get(0).getSql());
+    assertEquals(SELECT1.getSql(), requests.get(1).getSql());
+    assertTrue(requests.get(1).getTransaction().hasSingleUse());
+    assertEquals(INVALID_SELECT.getSql(), requests.get(2).getSql());
+    assertTrue(requests.get(2).getTransaction().hasSingleUse());
+    assertEquals(SELECT2.getSql(), requests.get(3).getSql());
+    assertTrue(requests.get(3).getTransaction().hasSingleUse());
+
+    List<CommitRequest> commitRequests = mockSpanner.getRequestsOfType(CommitRequest.class);
+    assertEquals(1, commitRequests.size());
+
+    List<RollbackRequest> rollbackRequests = mockSpanner.getRequestsOfType(RollbackRequest.class);
+    assertEquals(0, rollbackRequests.size());
   }
 
   @Test
-  public void testDdl() throws SQLException {
-    String sql = "CREATE TABLE foo (id bigint primary key)";
+  public void testTwoQueries() throws SQLException {
+    try (Connection connection = DriverManager.getConnection(createUrl())) {
+      try (java.sql.Statement statement = connection.createStatement()) {
+        // Statement#execute(String) returns true if the result is a result set.
+        assertTrue(statement.execute("SELECT 1; SELECT 2;"));
+
+        try (ResultSet resultSet = statement.getResultSet()) {
+          assertTrue(resultSet.next());
+          assertEquals(1L, resultSet.getLong(1));
+          assertFalse(resultSet.next());
+        }
+
+        // getMoreResults() returns true if the next result is a ResultSet.
+        assertTrue(statement.getMoreResults());
+        try (ResultSet resultSet = statement.getResultSet()) {
+          assertTrue(resultSet.next());
+          assertEquals(2L, resultSet.getLong(1));
+          assertFalse(resultSet.next());
+        }
+
+        // getMoreResults() should now return false. We should also check getUpdateCount() as that
+        // method should return -1 to indicate that there is also no update count available.
+        assertFalse(statement.getMoreResults());
+        assertEquals(-1, statement.getUpdateCount());
+      }
+    }
 
+    // The server will also receive a 'SELECT 1' statement from PgAdapter, as PgAdapter calls
+    // connection.isAlive() before using it.
+    List<ExecuteSqlRequest> requests = mockSpanner.getRequestsOfType(ExecuteSqlRequest.class);
+    assertEquals(2, requests.size());
+    assertEquals(SELECT1.getSql(), requests.get(0).getSql());
+    assertTrue(requests.get(0).getTransaction().hasBegin());
```

Comment:
This turns out to be incorrect (my fault, I had written that wrong in the design doc): A batch that is executed without an active transaction should execute each statement in auto-commit mode. So this select statement (and also the next one) should be executed using a single-use read-only transaction (or more simply: The `com.google.cloud.spanner.Connection` should be in auto-commit mode, and will handle that automatically for you)

Reply:
Done.